gordo_components/machine/dataset/datasets.py

```diff
@@ -122,11 +138,17 @@ def __init__(
         """
         self.train_start_date = self._validate_dt(train_start_date)
         self.train_end_date = self._validate_dt(train_end_date)
+
+        if self.train_start_date >= self.train_end_date:
+            raise ValueError(
+                f"train_end_date ({self.train_end_date}) must be after train_start_date ({self.train_start_date})"
+            )
+
         self.tag_list = normalize_sensor_tags(list(tag_list), asset, default_asset)
         self.target_tag_list = (
             normalize_sensor_tags(list(target_tag_list), asset, default_asset)
             if target_tag_list
-            else []
+            else self.tag_list.copy()
```

Comment:
:man_shrugging: yaaa. This is mostly because by combining the config `Dataset` and the associated validators, it would error if this was empty. We could copy a bit the logic of the model side and set `_strict=False` but I dunno, certainly seems like an edge case.

Reply:
I see, I have no particular opinion about it, was more a general wondering :man_shrugging: